utils/cache.js

```diff
@@ -0,0 +1,113 @@
+const CACHEEXPIRYTIME = 2; // Default cache expriy time in Minutes
+const CACHESIZE = 10; // Default maximum size for cache pool (in Megabytes)
+const minutesToMilliseconds = (minutes) => minutes * 60000; // Converts given minutes into milliseconds
+
+/**
+ * Cache pool to get and store API responses
+ * @param {Object} [opt] options for cache pool
+ * @param {number} opt.maximumSize Maximum size of the cache pool in Megabytes (MB)
+ */
+const cachePool = (opt = { maximumSize: CACHESIZE }) => {
+  const cacheStore = new Map();
+  let hits = 0;
+
+  /**
+   * Get an API Respnose from cacheStore.
+   * @param {string} key
+   * @returns {null | object}
+   */
+  const get = (key) => {
+    const apiData = cacheStore.get(key);
+
+    if (!apiData) {
+      return null;
+    }
+
+    const isApiDataExpired = new Date().getTime() > apiData.expiry;
+
+    // If data is expired remove it from store, time to get a fresh copy.
+    if (isApiDataExpired) {
+      cacheStore.delete(key);
+      return null;
+    }
+
+    hits += 1;
+    return apiData.response;
+  };
+
+  /**
+   * Add API response to cacheStore.
+   * @param {string} key
+   * @param {Object} value Value to be stored inside the cache.
+   * @param {number} value.priority Priority of the api
+   * @param {number} value.expiry Expiry time of api
+   * @param {number} value.size Size of api response in byte
+   * @returns {number} : statusCode
+   */
+  const set = async (key, value) => {
+    try {
+      cacheStore.set(key, value);
+      return 200;
+    } catch (err) {
+      return 500;
+    }
+  };
```

Comment:
1. Our cache seems to be tightly coupled with HTTP response codes. Can we make it more generic? Ideally it should not return anything
2. Why are we returning 500 response code if we are unable to set cache? This would mean that even if we are successful in fetching data from our db but fail to set it in cache we'll communicate a failure to the client.